### PR TITLE
Function day_of_week()

### PR DESCRIPTION
**Pull request by st-pasha**
New function `day_of_week()` that  computes the week-day component of a given date.

WIP for #2858

### CODE REVIEW

File: src/core/expr/time/fexpr_day_of_week.cc [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      }
+      return Column(new DayOfWeek_ColumnImpl(std::move(col)));
+    }
+};
+
+
+
+
+//------------------------------------------------------------------------------
+// Python-facing `ymd()` function
```
Comment (oleksiyskononenko) [inline src/core/expr/time/fexpr_day_of_week.cc:101]: `day_of_week` :)

File: src/core/lib/hh/date.cc [hunk 2]
```diff
@@ -104,12 +104,18 @@ int last_day_of_month(int y, int m) noexcept {
...
 //   z is number of days since 1970-01-01 and is in the range:
 //   [numeric_limits<int>::min(), numeric_limits<int>::max()-4].
-int weekday_from_days(int z) noexcept {
-  return (z >= -4) ? (z+4) % 7 : (z+5) % 7 + 6;
+int iso_weekday_from_days(int z) noexcept {
+  return (z >= -3) ? (z+3) % 7 + 1 : (z+4) % 7 + 7;
+}
+int bible_weekday_from_days(int z) noexcept {
+  return (z >= -4) ? (z+4) % 7 + 1 : (z+5) % 7 + 7;
 }
```
Comment (oleksiyskononenko) [inline src/core/lib/hh/date.cc:119]: Are you planning to use this function at some point?

**Review by oleksiyskononenko — APPROVED** [review]

File: src/core/lib/hh/date.cc (hunk 2 above)
Comment (st-pasha) [reply src/core/lib/hh/date.cc:119]: We might add it, as an alternative. I'm not sure at this point.

File: src/core/expr/time/fexpr_day_of_week.cc (hunk 1 above)
Comment (st-pasha) [reply src/core/expr/time/fexpr_day_of_week.cc]: lol, i did it again ;)

**Pull request merged**